enterprise/api/v1/views/default_enterprise_enrollments.py

```diff
@@ -211,6 +211,12 @@ def _partition_default_enrollment_intentions_by_enrollment_status(
                 already_enrolled.append((intention, non_audit_enrollment))
                 continue
 
+            if non_audit_enrollment and non_audit_enrollment.unenrolled:
```

Comment:
[sanity check] If the existing enrollment is audit (vs. the `non_audit_enrollment` here), the learners unenrolls, and there's non-audit modes that exist, would we also want to prevent automatic re-enrollment, too?

For example, would the conditions below the `NOTE` comment also need to take into account `audit_enrollment.unenrolled`?

Reply:
I thought about that but decided against it, I can leave a comment in the code. My reasoning is that default enrollments should be "exactly once" per (user, enterprise), if possible. If they enrolled in audit and unenrolled, it means they likely never had a default intention realized for them. This isn't a strongly-held position though.

Reply:
> If they enrolled in audit and unenrolled, it means they likely never had a default intention realized for them.

Ah yeah, this probably represents the source of most audit enrollments. The exception would be for courses that are audit-only, which likely rarely happens in production; it does happen somewhat frequently on stage, though we shouldn't optimize for bad stage data 😉 

Let's leave it as is for now, with a comment!